### PR TITLE
Fix inference causing existing node locations to be changed

### PR DESCRIPTION
**Pull request by brycepg**
Only set the inferred node lineno and col_offset if they are unset upon
inference.

Fixes PyCQA/pylint#1839

### CODE REVIEW

**Comment by brycepg** [comment]
These CI tools are getting really flakey

**Comment by brycepg** [comment]
AppVeyor is failing with

```
Command exited with code 1
dir "env:
```

File: astroid/tests/unittest_inference.py [hunk 1]
```diff
@@ -4273,7 +4291,6 @@ def test_duplicated_keyword_arguments(self):
         self.assertIn('f', site.duplicated_keywords)
 
 
-@unittest.expectedFailure
```
Comment (PCManticore) [inline astroid/tests/unittest_inference.py]: Why was this removed? I'm not sure we fixed this.

Comment (brycepg) [reply astroid/tests/unittest_inference.py]: I was trying to stop TravisCI from failing. I'll drop that line and re-push

**Comment by brycepg** [comment]
For some reason #472 is now affecting the Travis CI build

**Comment by PCManticore** [comment]
Interesting. I restarted the jobs to see if they work now. I imagine this might be caused by non-deterministic runs of the transforms, that means if inference transforms are suddenly applied in a different order than what was happening until now, we might see the original problem of the issue you linked gone (but as I mentioned there or in some other issue, if I recall correctly, the problem is that only one inference tip gets successful if we have multiple for the same node class, since the latter gets reset)

**Comment by PCManticore** [comment]
This is the issue I was thinking about: https://github.com/PyCQA/astroid/issues/444. Anyway, we can remove the expectedfailure decorator for now. Can you do that if you get a chance? also don't forget to rebase this branch so we can merge it.

**Comment by brycepg** [comment]
Thanks for the review @PCManticore, I've removed the expected failure. AppVeyor is acting up again.

I'll try to work on inference_tip stacking, I don't know much about it yet however.

**Comment by brycepg** [comment]
So it looks like this expected failure test is passing on TravisCI and Failing on AppVeyor.....

**Comment by brycepg** [comment]
I'll try to work on inference tip stacking

**Comment by PCManticore** [comment]
Thank you @brycepg ! Let's merge this, we can revisit the inference stacking afterwards. Let me know if I can help you with anything regarding that.
